2025-04-07 19:26:53,978 | INFO | TradingLogger initialized
2025-04-07 19:26:55,883 | INFO | METRIC training_start=1
2025-04-07 19:26:56,686 | ERROR | ERROR in system_run: could not broadcast input array from shape (5,35) into shape (5,3)
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 56, in _train
    model1.learn(total_timesteps=100000)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/ppo/ppo.py", line 315, in learn
    return super().learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/on_policy_algorithm.py", line 264, in learn
    total_timesteps, callback = self._setup_learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/base_class.py", line 423, in _setup_learn
    self._last_obs = self.env.reset()  # type: ignore[assignment]
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py", line 46, in reset
    observations = self.venv.reset()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 78, in reset
    self._save_obs(env_idx, obs)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 110, in _save_obs
    self.buf_obs[key][env_idx] = obs[key]  # type: ignore[call-overload]
ValueError: could not broadcast input array from shape (5,35) into shape (5,3)
2025-04-07 19:31:12,814 | INFO | TradingLogger initialized
2025-04-07 19:31:14,524 | INFO | METRIC training_start=1
2025-04-07 19:31:15,293 | ERROR | ERROR in system_run: could not broadcast input array from shape (5,35) into shape (5,3)
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 56, in _train
    model1.learn(total_timesteps=100000)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/ppo/ppo.py", line 315, in learn
    return super().learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/on_policy_algorithm.py", line 264, in learn
    total_timesteps, callback = self._setup_learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/base_class.py", line 423, in _setup_learn
    self._last_obs = self.env.reset()  # type: ignore[assignment]
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py", line 46, in reset
    observations = self.venv.reset()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 78, in reset
    self._save_obs(env_idx, obs)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 110, in _save_obs
    self.buf_obs[key][env_idx] = obs[key]  # type: ignore[call-overload]
ValueError: could not broadcast input array from shape (5,35) into shape (5,3)
2025-04-07 19:33:02,634 | INFO | TradingLogger initialized
2025-04-07 19:33:04,318 | INFO | METRIC training_start=1
2025-04-07 19:33:04,988 | ERROR | ERROR in system_run: could not broadcast input array from shape (5,35) into shape (5,3)
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 56, in _train
    model1.learn(total_timesteps=100000)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/ppo/ppo.py", line 315, in learn
    return super().learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/on_policy_algorithm.py", line 264, in learn
    total_timesteps, callback = self._setup_learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/base_class.py", line 423, in _setup_learn
    self._last_obs = self.env.reset()  # type: ignore[assignment]
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py", line 46, in reset
    observations = self.venv.reset()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 78, in reset
    self._save_obs(env_idx, obs)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 110, in _save_obs
    self.buf_obs[key][env_idx] = obs[key]  # type: ignore[call-overload]
ValueError: could not broadcast input array from shape (5,35) into shape (5,3)
2025-04-07 19:50:37,338 | INFO | TradingLogger initialized
2025-04-07 19:50:39,183 | INFO | METRIC training_start=1
2025-04-07 19:50:39,843 | ERROR | ERROR in system_run: could not broadcast input array from shape (5,35) into shape (5,3)
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 56, in _train
    model1.learn(total_timesteps=100000)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/ppo/ppo.py", line 315, in learn
    return super().learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/on_policy_algorithm.py", line 264, in learn
    total_timesteps, callback = self._setup_learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/base_class.py", line 423, in _setup_learn
    self._last_obs = self.env.reset()  # type: ignore[assignment]
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py", line 46, in reset
    observations = self.venv.reset()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 78, in reset
    self._save_obs(env_idx, obs)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 110, in _save_obs
    self.buf_obs[key][env_idx] = obs[key]  # type: ignore[call-overload]
ValueError: could not broadcast input array from shape (5,35) into shape (5,3)
2025-04-07 20:14:12,342 | INFO | TradingLogger initialized
2025-04-07 20:14:14,318 | INFO | METRIC training_start=1
2025-04-07 20:14:15,000 | ERROR | ERROR in system_run: could not broadcast input array from shape (5,35) into shape (5,3)
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 56, in _train
    model1.learn(total_timesteps=100000)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/ppo/ppo.py", line 315, in learn
    return super().learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/on_policy_algorithm.py", line 264, in learn
    total_timesteps, callback = self._setup_learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/base_class.py", line 423, in _setup_learn
    self._last_obs = self.env.reset()  # type: ignore[assignment]
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py", line 46, in reset
    observations = self.venv.reset()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 78, in reset
    self._save_obs(env_idx, obs)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 110, in _save_obs
    self.buf_obs[key][env_idx] = obs[key]  # type: ignore[call-overload]
ValueError: could not broadcast input array from shape (5,35) into shape (5,3)
2025-04-07 20:14:50,632 | INFO | TradingLogger initialized
2025-04-07 20:14:52,393 | INFO | METRIC training_start=1
2025-04-07 20:14:53,009 | ERROR | ERROR in system_run: could not broadcast input array from shape (5,35) into shape (5,3)
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 56, in _train
    model1.learn(total_timesteps=100000)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/ppo/ppo.py", line 315, in learn
    return super().learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/on_policy_algorithm.py", line 264, in learn
    total_timesteps, callback = self._setup_learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/base_class.py", line 423, in _setup_learn
    self._last_obs = self.env.reset()  # type: ignore[assignment]
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py", line 46, in reset
    observations = self.venv.reset()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 78, in reset
    self._save_obs(env_idx, obs)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 110, in _save_obs
    self.buf_obs[key][env_idx] = obs[key]  # type: ignore[call-overload]
ValueError: could not broadcast input array from shape (5,35) into shape (5,3)
2025-04-07 20:25:15,197 | INFO | TradingLogger initialized
2025-04-07 20:25:16,931 | INFO | METRIC training_start=1
2025-04-07 20:57:54,421 | INFO | TradingLogger initialized
2025-04-07 20:57:55,833 | INFO | METRIC training_start=1
2025-04-07 21:04:10,361 | INFO | TradingLogger initialized
2025-04-07 21:04:12,236 | INFO | METRIC training_start=1
2025-04-07 21:06:35,693 | INFO | TradingLogger initialized
2025-04-07 21:06:37,430 | INFO | METRIC training_start=1
2025-04-07 21:14:18,537 | INFO | TradingLogger initialized
2025-04-07 21:14:20,457 | INFO | METRIC training_start=1
2025-04-07 21:14:21,103 | ERROR | ERROR in system_run: could not broadcast input array from shape (5,16) into shape (5,3)
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 56, in _train
    model1.learn(total_timesteps=100000)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/ppo/ppo.py", line 315, in learn
    return super().learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/on_policy_algorithm.py", line 264, in learn
    total_timesteps, callback = self._setup_learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/base_class.py", line 423, in _setup_learn
    self._last_obs = self.env.reset()  # type: ignore[assignment]
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py", line 46, in reset
    observations = self.venv.reset()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 78, in reset
    self._save_obs(env_idx, obs)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 110, in _save_obs
    self.buf_obs[key][env_idx] = obs[key]  # type: ignore[call-overload]
ValueError: could not broadcast input array from shape (5,16) into shape (5,3)
2025-04-07 21:24:00,963 | INFO | TradingLogger initialized
2025-04-07 21:24:02,708 | INFO | METRIC training_start=1
2025-04-07 21:24:03,366 | ERROR | ERROR in system_run: could not broadcast input array from shape (5,19) into shape (5,3)
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 56, in _train
    model1.learn(total_timesteps=100000)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/ppo/ppo.py", line 315, in learn
    return super().learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/on_policy_algorithm.py", line 264, in learn
    total_timesteps, callback = self._setup_learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/base_class.py", line 423, in _setup_learn
    self._last_obs = self.env.reset()  # type: ignore[assignment]
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py", line 46, in reset
    observations = self.venv.reset()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 78, in reset
    self._save_obs(env_idx, obs)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 110, in _save_obs
    self.buf_obs[key][env_idx] = obs[key]  # type: ignore[call-overload]
ValueError: could not broadcast input array from shape (5,19) into shape (5,3)
2025-04-07 21:27:54,167 | INFO | TradingLogger initialized
2025-04-07 21:27:55,906 | INFO | METRIC training_start=1
2025-04-07 21:27:56,561 | ERROR | ERROR in system_run: could not broadcast input array from shape (5,19) into shape (5,3)
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 56, in _train
    model1.learn(total_timesteps=100000)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/ppo/ppo.py", line 315, in learn
    return super().learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/on_policy_algorithm.py", line 264, in learn
    total_timesteps, callback = self._setup_learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/base_class.py", line 423, in _setup_learn
    self._last_obs = self.env.reset()  # type: ignore[assignment]
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py", line 46, in reset
    observations = self.venv.reset()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 78, in reset
    self._save_obs(env_idx, obs)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 110, in _save_obs
    self.buf_obs[key][env_idx] = obs[key]  # type: ignore[call-overload]
ValueError: could not broadcast input array from shape (5,19) into shape (5,3)
2025-04-07 21:31:59,825 | INFO | TradingLogger initialized
2025-04-07 21:32:01,764 | INFO | METRIC training_start=1
2025-04-07 21:32:02,415 | ERROR | ERROR in system_run: could not broadcast input array from shape (5,19) into shape (5,3)
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 56, in _train
    model1.learn(total_timesteps=100000)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/ppo/ppo.py", line 315, in learn
    return super().learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/on_policy_algorithm.py", line 264, in learn
    total_timesteps, callback = self._setup_learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/base_class.py", line 423, in _setup_learn
    self._last_obs = self.env.reset()  # type: ignore[assignment]
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py", line 46, in reset
    observations = self.venv.reset()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 78, in reset
    self._save_obs(env_idx, obs)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 110, in _save_obs
    self.buf_obs[key][env_idx] = obs[key]  # type: ignore[call-overload]
ValueError: could not broadcast input array from shape (5,19) into shape (5,3)
2025-04-07 21:43:06,516 | INFO | TradingLogger initialized
2025-04-07 21:43:07,819 | ERROR | ERROR in system_run: 'numpy.ndarray' object has no attribute 'columns'
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 46, in _train
    env = InstitutionalTradingEnv(symbol_data)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/env/trading_env.py", line 23, in __init__
    self.observation_space = self._init_observation_space()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/env/trading_env.py", line 79, in _init_observation_space
    n_features = len(sample_features.columns)
AttributeError: 'numpy.ndarray' object has no attribute 'columns'
2025-04-07 21:46:06,931 | INFO | TradingLogger initialized
2025-04-07 21:46:08,824 | ERROR | ERROR in system_run: 'numpy.ndarray' object has no attribute 'columns'
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 46, in _train
    env = InstitutionalTradingEnv(symbol_data)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/env/trading_env.py", line 23, in __init__
    self.observation_space = self._init_observation_space()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/env/trading_env.py", line 79, in _init_observation_space
    n_features = len(sample_features.columns)
AttributeError: 'numpy.ndarray' object has no attribute 'columns'
2025-04-07 21:47:26,536 | INFO | TradingLogger initialized
2025-04-07 21:47:28,130 | ERROR | ERROR in system_run: 'numpy.ndarray' object has no attribute 'columns'
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 46, in _train
    env = InstitutionalTradingEnv(symbol_data)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/env/trading_env.py", line 23, in __init__
    self.observation_space = self._init_observation_space()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/env/trading_env.py", line 79, in _init_observation_space
    n_features = len(sample_features.columns)
AttributeError: 'numpy.ndarray' object has no attribute 'columns'
2025-04-07 21:50:13,232 | INFO | TradingLogger initialized
2025-04-07 21:50:15,284 | INFO | METRIC training_start=1
2025-04-07 21:55:44,228 | INFO | TradingLogger initialized
2025-04-07 21:55:44,241 | INFO | METRIC training_start=1
2025-04-07 21:58:19,623 | INFO | TradingLogger initialized
2025-04-07 21:58:19,635 | INFO | METRIC training_start=1
2025-04-07 22:10:18,599 | ERROR | ERROR in system_run: cannot pickle 'mappingproxy' object
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 72, in _train
    self._save_agent()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 143, in _save_agent
    pickle.dump(self.agent, f)
TypeError: cannot pickle 'mappingproxy' object
2025-04-07 22:13:43,522 | INFO | TradingLogger initialized
2025-04-07 22:13:43,533 | ERROR | ERROR in system_run: 'InstitutionalFeatureEngineer' object has no attribute 'min_data_points'
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 46, in _train
    env = InstitutionalTradingEnv(symbol_data)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/env/trading_env.py", line 32, in __init__
    self.reset()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/env/trading_env.py", line 110, in reset
    self.current_step = self.feature_engineer.min_data_points
AttributeError: 'InstitutionalFeatureEngineer' object has no attribute 'min_data_points'
2025-04-07 22:17:09,860 | INFO | TradingLogger initialized
2025-04-07 22:17:11,492 | INFO | METRIC training_start=1
2025-04-07 22:34:14,671 | ERROR | ERROR in system_run: cannot pickle 'mappingproxy' object
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 72, in _train
    self._save_agent()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 143, in _save_agent
    pickle.dump(self.agent, f)
TypeError: cannot pickle 'mappingproxy' object
2025-04-07 22:55:35,702 | INFO | TradingLogger initialized
2025-04-07 22:55:37,505 | ERROR | ERROR in system_run: 'InstitutionalTradingEnv' object has no attribute 'max_position_size'
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 46, in _train
    env = InstitutionalTradingEnv(symbol_data)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/env/trading_env.py", line 31, in __init__
    self.action_space = self._init_action_space()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/env/trading_env.py", line 70, in _init_action_space
    low=np.array([-self.max_position_size, 0, 0] * self.n_symbols),
AttributeError: 'InstitutionalTradingEnv' object has no attribute 'max_position_size'
2025-04-07 22:59:23,551 | INFO | TradingLogger initialized
2025-04-07 22:59:23,568 | INFO | METRIC training_start=1
2025-04-07 22:59:24,255 | ERROR | ERROR in system_run: 'InstitutionalTradingEnv' object has no attribute '_get_current_price'
Traceback (most recent call last):
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 33, in run
    self._train()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/system.py", line 56, in _train
    model1.learn(total_timesteps=100000)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/ppo/ppo.py", line 315, in learn
    return super().learn(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/on_policy_algorithm.py", line 277, in learn
    continue_training = self.collect_rollouts(self.env, callback, self.rollout_buffer, n_rollout_steps=self.n_steps)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/on_policy_algorithm.py", line 194, in collect_rollouts
    new_obs, rewards, dones, infos = env.step(clipped_actions)
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/base_vec_env.py", line 206, in step
    return self.step_wait()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/vec_check_nan.py", line 40, in step_wait
    observations, rewards, dones, infos = self.venv.step_wait()
  File "/Users/benjaminigbokwe/projects/qprotraderV4/myenv/lib/python3.9/site-packages/stable_baselines3/common/vec_env/dummy_vec_env.py", line 58, in step_wait
    obs, self.buf_rews[env_idx], terminated, truncated, self.buf_infos[env_idx] = self.envs[env_idx].step(
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/env/trading_env.py", line 115, in step
    self._execute_trade(symbol, action[i])
  File "/Users/benjaminigbokwe/projects/qprotraderV4/core/env/trading_env.py", line 135, in _execute_trade
    price = self._get_current_price(symbol)
AttributeError: 'InstitutionalTradingEnv' object has no attribute '_get_current_price'
2025-04-07 23:08:40,809 | INFO | TradingLogger initialized
2025-04-07 23:08:40,827 | INFO | METRIC training_start=1
2025-04-07 23:24:37,643 | INFO | TradingLogger initialized
2025-04-07 23:24:39,300 | INFO | METRIC training_start=1
